File: Clases-Bluej/Producto.java
public class Producto {
    private String idProducto;
    private String nombre;
    private int cantidadStock;
    private double precioUnitario;

    public Producto(String idProducto, String nombre, int cantidadStock, double precioUnitario) {
        this.idProducto = idProducto;
        this.nombre = nombre;
        this.cantidadStock = cantidadStock;
        this.precioUnitario = precioUnitario;
    }

    public void reabastecer(int cantidad) {
        this.cantidadStock += cantidad;
    }

    public boolean reducirStock(int cantidad) {
        if (this.cantidadStock >= cantidad) {
            this.cantidadStock -= cantidad;
            return true;
        }
        return false;
    }

    public String getIdProducto() { return idProducto; }
    public String getNombre() { return nombre; }
    public int getCantidadStock() { return cantidadStock; }
    public double getPrecioUnitario() { return precioUnitario; }
}

File: Clases_Csharp/Producto.cs
namespace SistemaVeterinaria
{
    public class Producto
    {
        public string IdProducto { get; private set; }
        public string Nombre { get; private set; }
        public int CantidadStock { get; private set; }
        public double PrecioUnitario { get; private set; }

        public Producto(string idProducto, string nombre, int cantidadStock, double precioUnitario)
        {
            IdProducto = idProducto;
            Nombre = nombre;
            CantidadStock = cantidadStock;
            PrecioUnitario = precioUnitario;
        }

        public void Reabastecer(int cantidad)
        {
            CantidadStock += cantidad;
        }

        public bool ReducirStock(int cantidad)
        {
            if (CantidadStock >= cantidad)
            {
                CantidadStock -= cantidad;
                return true;
            }
            return false;
        }
    }
}